### PR TITLE
Output fractions of each vegetation/soil type category

### LINKED ISSUE #709: Update sfc_climo_gen to output fractional vegetation and soil type
Currently, this program outputs a dominate category at each model point. Update the program to output the percentage of each category within a model point.

### PR DESCRIPTION
**Pull request by GeorgeGayno-NOAA**
## DESCRIPTION OF CHANGES: 
Update sfc_climo_gen to optionally output the fraction of each vegetation and soil
type category at a model grid point.

## TESTS CONDUCTED: 
New unit test and consistency tests were added. 

The branch at f527005 was checked out on Hera, Jet, Orion and WCOSS2.

Hera:
- Successfully compiled using Intel/GNU (both 'Release'/'Debug' build types)
- Doxygen successfully built. 
- Unit tests successfully run locally.
- The `grid_gen` consistency tests all passed as expected (because these changes are backwards compatible).

WCOSS2:
 - Successfully compiled using Intel (both 'Release' and 'Debug' build types).
 - The `grid_gen` consistency tests all passed.

Orion and Jet:
 - Successfully compiled using Intel ('Release' build type).
 - The `grid_gen` consistency tests all passed.
 
## DEPENDENCIES:
Attributes were added to the input soil and vegetation type files and were hosted by @KateFriedman-NOAA.

## DOCUMENTATION:
Updated for changes: https://georgegayno-noaaufs-utils.readthedocs.io/en/feature-sfc_climo_gen.frac/ufs_utils.html#sfc-climo-gen

## ISSUE: 
Fixes #709.

### CODE REVIEW

**Comment by GeorgeGayno-NOAA** [comment]
@barlage Who in the land group could run their own tests?

**Comment by KateFriedman-NOAA** [comment]
> Attributes were added to the input soil and vegetation type files. Ask @KateFriedman-NOAA to host them before merging.

Have I done this for you already or are there new files you're ready for me to pull in now? Let me know!

**Review by BenjaminBlake-NOAA — APPROVED** [review]
I tested these changes on Hera and successfully created the surface climo fix files with the fractional vegetation and soil information.

File: reg_tests/grid_gen/driver.hera.sh [hunk 1]
```diff
@@ -86,27 +86,36 @@ TEST3=$(sbatch --parsable --ntasks-per-node=24 --nodes=1 -t 0:07:00 -A $PROJECT_
       -o $LOG_FILE3 -e $LOG_FILE3 ./gfdl.regional.sh)
 
 #-----------------------------------------------------------------------------
-# esg regional grid
+# ESG regional grid (output dominate soil/vegetation type).
```
Comment (JeffBeck-NOAA) [inline reg_tests/grid_gen/driver.hera.sh]: Dominant?  Same comment for change below in this file and in other files.

File: cmake/mpiexec.hera [hunk 2]
```diff
@@ -6,7 +6,7 @@
 #   $2+ - Executable and its arguments
 #
 
-ACCOUNT=gsd-fv3
+ACCOUNT=fv3-cpu
 QOS=debug
```
Comment (JeffBeck-NOAA) [inline cmake/mpiexec.hera]: Should this change be committed?

File: reg_tests/grid_gen/driver.hera.sh (hunk 1 above)
Comment (GeorgeGayno-NOAA) [reply reg_tests/grid_gen/driver.hera.sh]: Thank you. I can't spell.

File: cmake/mpiexec.hera (hunk 2 above)
Comment (GeorgeGayno-NOAA) [reply cmake/mpiexec.hera]: Reverted to original setting.

**Review by JeffBeck-NOAA — APPROVED** [review]

**Comment by GeorgeGayno-NOAA** [comment]
Will merge, then add the baseline files for the new consistency test.

**Pull request merged**